src/mixer.cpp

```diff
@@ -132,55 +680,192 @@ void Mixer::set_new_aux_command(aux_command_t new_aux_command)
   }
 }
 
-void Mixer::mix_output()
+float Mixer::mix_multirotor_without_motor_parameters(Controller::Output commands)
 {
-  Controller::Output commands = RF_.controller_.output();
-  float max_output = 1.0f;
+  // Mix the inputs
+  float max_output = 1.0;
 
-  // Reverse fixed-wing channels just before mixing if we need to
-  if (RF_.params_.get_param_int(PARAM_FIXED_WING)) {
-    commands.x *= RF_.params_.get_param_int(PARAM_AILERON_REVERSE) ? -1 : 1;
-    commands.y *= RF_.params_.get_param_int(PARAM_ELEVATOR_REVERSE) ? -1 : 1;
-    commands.z *= RF_.params_.get_param_int(PARAM_RUDDER_REVERSE) ? -1 : 1;
-  } else if (commands.F < RF_.params_.get_param_float(PARAM_MOTOR_IDLE_THROTTLE)) {
-    // For multirotors, disregard yaw commands if throttle is low to prevent motor spin-up while
-    // arming/disarming
-    commands.z = 0.0;
+  for (uint8_t i = 0; i < NUM_MIXER_OUTPUTS; i++) {
+    if ((*mixer_to_use_.output_type)[i] != NONE) {
+      // Matrix multiply to mix outputs
+      outputs_[i] = commands.Fx * (*mixer_to_use_.Fx)[i] +
+                    commands.Fy * (*mixer_to_use_.Fy)[i] +
+                    commands.Fz * (*mixer_to_use_.Fz)[i] + 
+                    commands.Qx * (*mixer_to_use_.Qx)[i] + 
+                    commands.Qy * (*mixer_to_use_.Qy)[i] + 
+                    commands.Qz * (*mixer_to_use_.Qz)[i];
+
+      // Save off the largest control output (for motors) if it is greater than 1.0 for future scaling
+      if ((*mixer_to_use_.output_type)[i] == M) {
+        if (abs(outputs_[i]) > max_output) { max_output = abs(outputs_[i]); }
+      }
+    }
   }
 
-  if (mixer_to_use_ == nullptr) { return; }
```

Comment:
Should we keep this check around?

Reply:
No, it couldn't be called. The check lives at the top of the `mix_output` method, where it won't run the mixer if the mixer isn't selected. This check is still here, but not at line 705. I think GitHub jumbled the lines because I added a bunch of new ones.